static/app/types/index.tsx

```diff
@@ -21,6 +22,23 @@ import {DynamicSamplingRules} from './dynamicSampling';
 import {Event} from './event';
 import {Mechanism, RawStacktrace, StacktraceType} from './stacktrace';
 
+export type OnSentryInitConfiguration =
+  | {
+      name: 'passwordStrength';
+      input: string;
+      element: string;
+    }
+  | {
+      name: 'renderReact';
+      component: 'Indicators' | 'SetupWizard' | 'SystemAlerts';
```

Comment:
Can we make this an explicit type so that we can type the `COMPONENT_MAP` using this, so adding new ones will break types and you'll have to update this?

Or inverse the relationship maybe and have this take the type of the constant `COMPONENT_MAP`. (might affect import order though.